test-harness/modules/mocker/dns.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (C) Mesosphere, Inc. See LICENSE file for details.
+
+"""
+Programmable in-memory DNS server
+"""
+
+import logging
+
+from dnslib import (
+    A,
+    DNSRecord,
+    DNSHeader,
+    QTYPE,
+    RR,
+)
+
+from dnslib.server import (
+    BaseResolver,
+    DNSLogger,
+    DNSServer,
+)
+
+log = logging.getLogger(__name__)
+
+
+class DomainName(str):
+    """DomainName is helper class that allow easier manipulation with DNS names.
+
+       Example:
+          com = DomainName("com")
+          assert com.example == "example.com"
+    """
+
+    def __getattr__(self, item):
+        return DomainName(item + '.' + self)
+
+
+class EmptyResolver(BaseResolver):
+    """EmptyResolver returns empty reply for any DNS query"""
+
+    def resolve(self, request, handler):
+        return DNSRecord(
+            DNSHeader(id=request.header.id, qr=1, aa=1, ra=1), q=request.q)
+
+
+class MesosLeaderResolver(BaseResolver):
+    """MesosLeaderResolver returns responses to leader.mesos queries"""
+
+    def __init__(self, leader_ip, ttl=5, domain=DomainName('mesos')):
+        self.leader_ip = leader_ip
+        self.ttl = ttl
+        self.domain = domain
+
+    def resolve(self, request, handler):
+        # We can respond only to leader.mesos A requests
+        if (str(request.q.qname) == (self.domain.leader + ".") and
```

Comment:
They are not used at the moment - true.

But the thing is that we want other people to write tests as well - let's make it easier for them. It costs us very little to do it, and it may save them some time if/when they need it.